ts/services/ElementService.ts

```diff
@@ -0,0 +1,172 @@
+import * as Bluebird from 'bluebird';
+import * as fs from 'fs';
+import * as request from 'request-promise';
+
+const MAINT_TAG_NAME = 'n.state.maintenance';
+const MAINT_END_TAG_NAME = 'n.state.maintenance_end';
+const MS_IN_HOUR = 3600000;
+
+class ElementService {
```

Comment:
If this class is exclusively for maintenance mode then `ElementMaintenanceService` seems more appropriate. Not much of this class is reusable for general element operations as is other than the tag methods.

Reply:
I don't it will just be for maint mode, add/remove tags and listing of elements at least would be easy to add here.